thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalyResource.java

```diff
@@ -580,69 +581,77 @@ public AnomalyTimelinesView getAnomalyMergedResultTimeSeries(@NotNull @PathParam
+      anomalyTimelinesView = AnomalyTimelinesView.fromJsonString(
+          anomalyProps.get("anomalyTimelinesView"));
+    } else {
+
+      DimensionMap dimensions = anomalyResult.getDimensions();
+      AnomalyFunctionDTO anomalyFunctionSpec = anomalyResult.getFunction();
+      BaseAnomalyFunction anomalyFunction = anomalyFunctionFactory.fromSpec(anomalyFunctionSpec);
+
+      // Calculate view window start and end if they are not given by the user, which should be 0 if it is not given.
+      // By default, the padding window size is half of the anomaly window.
+      if (viewWindowStartTime == 0 || viewWindowEndTime == 0) {
+        long anomalyWindowStartTime = anomalyResult.getStartTime();
+        long anomalyWindowEndTime = anomalyResult.getEndTime();
+
+        long bucketMillis = TimeUnit.MILLISECONDS.convert(anomalyFunctionSpec.getBucketSize(), anomalyFunctionSpec.getBucketUnit());
+        long bucketCount = (anomalyWindowEndTime - anomalyWindowStartTime) / bucketMillis;
+        long paddingMillis = Math.max(1, (bucketCount / 2)) * bucketMillis;
+        if (paddingMillis > TimeUnit.DAYS.toMillis(1)) {
+          paddingMillis = TimeUnit.DAYS.toMillis(1);
+        }
+
+        if (viewWindowStartTime == 0) {
+          viewWindowStartTime = anomalyWindowStartTime - paddingMillis;
+        }
+        if (viewWindowEndTime == 0) {
+          viewWindowEndTime = anomalyWindowEndTime + paddingMillis;
+        }
       }
-    }
 
-    TimeGranularity timeGranularity =
-        Utils.getAggregationTimeGranularity(aggTimeGranularity, anomalyFunctionSpec.getCollection());
-    long bucketMillis = timeGranularity.toMillis();
-    // ThirdEye backend is end time exclusive, so one more bucket is appended to make end time inclusive for frontend.
-    viewWindowEndTime += bucketMillis;
+      TimeGranularity timeGranularity = Utils.getAggregationTimeGranularity(aggTimeGranularity, anomalyFunctionSpec.getCollection());
+      long bucketMillis = timeGranularity.toMillis();
+      // ThirdEye backend is end time exclusive, so one more bucket is appended to make end time inclusive for frontend.
+      viewWindowEndTime += bucketMillis;
 
-    long maxDataTime = collectionMaxDataTimeCache.get(anomalyResult.getCollection());
-    if (viewWindowEndTime > maxDataTime) {
-      viewWindowEndTime =
-          (anomalyResult.getEndTime() > maxDataTime) ? anomalyResult.getEndTime() : maxDataTime;
-    }
+      long maxDataTime = collectionMaxDataTimeCache.get(anomalyResult.getCollection());
+      if (viewWindowEndTime > maxDataTime) {
+        viewWindowEndTime = (anomalyResult.getEndTime() > maxDataTime) ? anomalyResult.getEndTime() : maxDataTime;
+      }
 
-    AnomalyDetectionInputContext adInputContext = TimeBasedAnomalyMerger
-        .fetchDataByDimension(viewWindowStartTime, viewWindowEndTime, dimensions, anomalyFunction,
-            anomalyMergedResultDAO, overrideConfigDAO, false);
+      AnomalyDetectionInputContext adInputContext =
```

Comment:
Yes. I just add few lines here. In brief, I check if the AnomalyTimelines is stored in Properties. If yes, I take the result from Properties; otherwise, calculate the baseline.

    if(anomalyProps.containsKey("anomalyTimelinesView")) {
      anomalyTimelinesView = AnomalyTimelinesView.fromJsonString(
          anomalyProps.get("anomalyTimelinesView"));
    } else {
...
}